via-router/src/router.rs

```diff
@@ -223,22 +227,141 @@ where
+        "/admin/tag/:tag_id/edit",
+        "/admin/settings",
+        "/categories",
+        "/category/:category_id",
+        "/category/:category_id/posts",
+        "/tags",
+        "/tag/:tag_id",
+        "/tag/:tag_id/posts",
+        "/favorites",
+        "/favorite/:item_id",
+        "/friends",
+        "/friend/:friend_id",
+        "/groups",
+        "/group/:group_id",
+        "/group/:group_id/members",
+        "/group/:group_id/posts",
+        "/events",
+        "/event/:event_id",
+        "/event/:event_id/rsvp",
+        "/event/:event_id/attendees",
+        "/help",
+        "/help/article/:article_id",
+        "/terms",
+        "/privacy",
+        "/faq",
+        "/sitemap",
+        "/rss",
+        "/api/:version/:resource",
+        "/api/:version/:resource/:resource_id",
+        "/api/:version/:resource/:resource_id/edit",
+        "/api/:version/:resource/:resource_id/comments/:comment_id",
+        "/api/:version/:resource/:resource_id/comments/:comment_id/edit",
+        "/checkout",
+        "/checkout/cart",
+        "/checkout/payment",
+        "/checkout/confirmation",
+        "/dashboard",
+        "/dashboard/overview",
+        "/dashboard/stats",
+        "/dashboard/reports",
+        "/notifications/settings",
+        "/notifications/settings/email",
+        "/notifications/settings/include",
+        "/inbox",
+        "/inbox/:conversation_id",
+        "/inbox/:conversation_id/messages",
+        "/subscriptions",
+        "/subscription/:subscription_id",
+        "/subscription/:subscription_id/edit",
+        "/billing",
+        "/billing/history",
+        "/billing/payment-methods",
+        "/billing/invoice/:invoice_id",
+        "/report/user/:user_id",
+        "/report/post/:post_id",
+        "/report/comment/:comment_id",
+        "/invite",
     ];
 
+    fn simple_prng(seed: u32) -> u32 {
+        // Constants from Numerical Recipes
+        let a: u64 = 1664525;
+        let c: u64 = 1013904223;
+        let m: u64 = u32::MAX as u64 + 1;
+
+        let next = (a * seed as u64 + c) % m;
+        next as u32 % 10_000 + 1
+    }
+
     #[test]
     fn test_router_visit() {
         let mut router = Router::new();
 
-        for path in &PATHS {
-            let _ = router.at(path).respond(());
+        for pattern in PATHS {
+            let _ = router.at(pattern).respond(());
         }
 
-        println!("router: {:#?}", router);
+        for pattern in PATHS {
+            let path = crate::path::Split::new(pattern)
+                .map(|[start, end]| {
+                    let segment = &pattern[start..end];
+                    if segment.starts_with(':') {
+                        simple_prng(12345).to_string()
+                    } else {
+                        segment.to_owned()
+                    }
+                })
+                .fold(String::new(), |path, next| path + "/" + &next);
+
+            router.visit(&path);
+        }
 
         // {
```

Comment:
I'm going to fix the tests. Not tonight and perhaps not before the next release. Heads up, empty path segments are a thing now (not a bug). In my opinion, they are completely safe and preferable. In probably every scenario it results in the router returning early rather than assuming it's a typo.

Reply:
For now, I think using SmallVec for nodes to optimize for the case of a single match is sufficient...

Reply:
I also tried implementing a flat version of the results vec as we had before. I think keeping the tree structure in tact makes sense. Even if it's a little slower, I like the way the results match the structure of the router.